File: website/src/main/java/pl/wilenskid/api/model/bean/PostBean.java
package pl.wilenskid.api.model.bean;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import lombok.Data;
import org.springframework.core.io.Resource;

import java.util.List;

@Data
@JsonAutoDetect
public class PostBean {
  private Long id;
  private String title;
  private FileBean description;
  private List<UserBean> contributors;
  private List<SubPostBean> subPosts;
  private List<TagBean> tags;
  private String created;
  private String updated;
}

File: website/src/main/typescript/app/front-end/api/models/post-bean.ts
import { UserBean } from './user-bean';
import { SubPostBean } from './sub-post-bean';
import { TagBean } from './tag-bean';
import { FileBean } from './file-bean';

export interface PostBean {
	readonly id: number;
	readonly title: string;
	readonly description: FileBean;
	readonly contributors: UserBean[];
	readonly subPosts: SubPostBean[];
	readonly tags: TagBean[];
	readonly created: string;
	readonly updated: string;
}
